freeipa/src/main/java/com/sequenceiq/freeipa/service/image/CoreImageProvider.java

```diff
@@ -41,6 +51,29 @@
         }
     }
 
+    @Override
+    public List<ImageWrapper> getImages(ImageSettingsRequest imageSettings, String region, String platform) {
+        try {
+            ImagesV4Response imagesV4Response = imageCatalogV4Endpoint.getImagesByName(WORKSPACE_ID_DEFAULT, imageSettings.getCatalog(), null, platform, null,
+                    null);
+            LOGGER.debug("Images received: {}", imagesV4Response);
+            return Optional.ofNullable(imagesV4Response.getFreeipaImages()).
+                    orElseGet(List::of).stream()
+                    .map(this::convert)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
```

Comment:
Instead of these two lines I'd recommend to use `.flatMap(Optional::stream)`

Reply:
thanks